odoo/addons/base/models/ir_model.py

```diff
@@ -1146,7 +1146,7 @@ def unlink(self):
         for selection in self:
             if selection.field_id.store:
                 # replace the value by NULL in the field's corresponding column
-                query = "UPDATE {table} SET {field}=NULL WHERE {field}=%s".format(
+                query = 'UPDATE {table} SET "{field}"=NULL WHERE "{field}"=%s'.format(
```

Comment:
You can even properly quote the table identifier:
```suggestion
                query = 'UPDATE "{table}" SET "{field}"=NULL WHERE "{field}"=%s'.format(
```